src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -872,64 +922,112 @@ CreateDistributedExecution(RowModifyLevel modLevel, List *taskList, bool hasRetu
 
 
 /*
- * StartDistributedExecution sets up the coordinated transaction and 2PC for
- * the execution whenever necessary. It also keeps track of parallel relation
- * accesses to enforce restrictions that arise due to foreign keys to reference
- * tables.
+ * InitialiseCoordinatedTransactionForTaskList decides whether to initiate a coordinated
+ * transaction and whether to use 2PC for the given task list. These are global settings
+ * that affect the remainder of the transaction.
+ *
+ * Additionally, it returns whether the current execution should error out on any failure
+ * and whether to use transaction blocks, which may depend on previous statements in the
+ * transaction.
  */
-void
-StartDistributedExecution(DistributedExecution *execution)
+static void
+InitialiseCoordinatedTransactionForTaskList(RowModifyLevel modLevel, List *taskList,
```

Comment:
This function doesnt necessarily start a coordinated transaction, the method name implies that it starts a transaction always. This is more like Initializing the steps for execution, so I feel like we should choose a different name